diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/ASTNodeUtilsTest.kt

```diff
@@ -61,6 +75,439 @@ class ASTNodeUtilsTest {
+            }
+            if (node.getAllChildrenWithType(IDENTIFIER).isNotEmpty()) {
+                Assert.assertEquals(node.getAllChildrenWithType(IDENTIFIER)[0].text, node.getAllIdentifierChildren()[0].text)
+                Assert.assertEquals(node.getAllChildrenWithType(IDENTIFIER)[0].text, node.getIdentifierName()!!.text)
+            }
+        }
+    }
+
+    @Test
+    fun `test getFirstChildWithType`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if(node.getAllChildrenWithType(IDENTIFIER).isNotEmpty()){
+                Assert.assertEquals(node.getFirstChildWithType(IDENTIFIER)!!.text, node.getAllChildrenWithType(IDENTIFIER)[0].text)
+                Assert.assertEquals(node.getFirstChildWithType(IDENTIFIER)!!.text, node.getIdentifierName()!!.text)
+            }
+        }
+    }
+
+    @Test
+    fun `test hasChildOfType`(){
+        val code = """
+            class Test {
+                val x = 0
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.getIdentifierName() != null)
+                Assert.assertTrue(node.hasChildOfType(IDENTIFIER))
+        }
+    }
+
+    @Test
+    fun `test hasAnyChildOfTypes`(){
+        val code = """
+            class Test {
+                val x = 0
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.getAllChildrenWithType(IDENTIFIER).isNotEmpty() || node.getAllChildrenWithType(CLASS).isNotEmpty()) {
+                val param = arrayOf(IDENTIFIER, CLASS)
+                Assert.assertTrue(node.hasAnyChildOfTypes(*param))
+            }
+        }
+    }
+
+    @Test
+    fun `test findChildBefore`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.findChildBefore(CLASS_BODY, CLASS) != null)
+                Assert.assertEquals(node.findChildBefore(CLASS_BODY, CLASS)!!.text, code)
+        }
+    }
+
+    @Test
+    fun `test findChildAfter`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.findChildAfter(VALUE_PARAMETER_LIST, TYPE_REFERENCE) != null)
+                Assert.assertEquals(node.findChildAfter(VALUE_PARAMETER_LIST, TYPE_REFERENCE)!!.text, "Int")
+        }
+    }
+
+    @Test
+    fun `test allSiblings`(){
```

Comment:
Make name of this test more meaningful: you are actually testing that `withSelf = true` works as expected, not just how `allSiblings` method works.